src/lib/uTop_main.cppo.ml

```diff
@@ -183,6 +183,11 @@ class read_phrase ~term = object(self)
           let result = parse_and_check input eos_is_error in
           return_value <- Some result;
           LTerm_history.add UTop.history input;
+          ignore(match result with
+            | UTop.Value _, _ ->
+                LTerm_history.add UTop.history input;
```

Comment:
@diml I just noticed what you were talking about, and no it is not necessary to save the normal history again within the `ignore`.